ballista/scheduler/src/state/task_manager.rs

```diff
@@ -738,17 +639,18 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> TaskManager<T, U>
     }
 
     /// Get the `ExecutionGraph` for the given job ID from cache
-    pub(crate) async fn get_active_execution_graph(
+    pub(crate) fn get_active_execution_graph(
         &self,
         job_id: &str,
     ) -> Option<Arc<RwLock<ExecutionGraph>>> {
         self.active_job_cache
             .get(job_id)
-            .map(|value| value.execution_graph.clone())
+            .as_deref()
+            .map(|cached| cached.execution_graph.clone())
     }
 
     /// Remove the `ExecutionGraph` for the given job ID from cache
-    pub(crate) async fn remove_active_execution_graph(
```

Comment:
@thinkharderdev  Sorry to interrupte, plz take look, could you plz tell me why remove the `async `, same in other functions 🤔

Reply:
This function is just removing the `ExecutionGraph` from the local active cache so it doesn't need to be async. We only need an async function when we modify the global state.

Reply:
I think `active_job_cache ` here is a global states here in `TaskManager` ->  `SchedulerState`, Is there something wrong ?  plz tell me 🤔

Reply:
Sorry, by "global state" I mean state shared across multiple schedulers. The `active_job_cache` is local to a single scheduler and and a purely in-memory data structure